docs/getting-started/auth-self-defence-checks.adoc

```diff
@@ -0,0 +1,148 @@
+
+=== Invoking Multiple Self Defence Checks
+
+==== Synchronously
+
+To run multiple self defence checks _synchronously_, you'll first need to invoke the `SecurityService#getCheckExecutor` method.  This methods returns `SyncSecurityCheckExecutor` where checks can be added (chained) to the executor and executed synchronously.
+
+
+[source, java]
+----
+SyncSecurityCheckExecutor syncCheckExecutor = securityService.getCheckExecutor();
+----
+
+===== Custom Self Defence Checks
+
+To add custom self defence checks to the `syncCheckExecutor`, the `SyncSecurityCheckExecutor#addCheck` method must be used.  This method expects `SecurityCheck` as a parameter and returns `SyncSecurityCheckExecutor`.  
+
+[source, java]
+----
+class customSecurityCheck implements SecurityCheck {
+    /**
+     * Custom security check.
+     *
+     * @param context Context to be used by the check.
+     * @return <code>true</code> or <code>false</code>
+     */
+    @Override
+    public SecurityCheckResult test(final Context context) {
+        // implement security check in customSecurityCheck()
+        boolean result = customSecurityCheck();
+        return new SecurityCheckResultImpl(this, result);
+    }
+}
+
+// add custom self defence check to the syncCheckExecutor
+syncCheckExecutor.addCheck(new customSecurityCheck());
+----
+
+===== Provided Self Defence Checks
+
+To add self defence checks available in the SDK, you must invoke the `SyncSecurityCheckExecutor#addCheck` method.  This method expects  `SecurityCheckType` as a parameter and returns `SyncSecurityCheckExecutor`.
+
+
+[source, java]
+----
+// add two provided self defence checks to the syncCheckExecutor
+syncCheckExecutor.addCheck(SecurityCheckType.<check_type>).addCheck(SecurityCheckType.<check_type>);
+----
+
+To run all the security self defence checks that have been added to the `syncCheckExecutor` the `SyncSecurityCheckExecutor#execute` method must be invoked.  This method returns a `Map` where the key is of type `String` and the value is of type `SecurityCheckResult`.
+
+[source, java]
+----
+Map<String, SecurityCheckResult> results = syncCheckExecutor.execute();
+----
+
+The `SyncSecurityCheckExecutor` allows for multiple checks to be chained so the above can be refactored to:
+[source, java]
+----
+Map<String, SecurityCheckResult> results = securityService.getCheckExecutor().addCheck(new customSecurityCheck()).addCheck(SecurityCheckType.<check_type>).addCheck(SecurityCheckType.<check_type>).execute();
+----
+
+==== Asynchronously
+
+To run multiple self defence checks _asynchronously_, you’ll first need to invoke the `SecurityService#getAsyncCheckExecutor` method. This methods returns `AsyncSecurityCheckExecutor` where checks can be added (chained) to the executor and executed asynchronously.
+
+[source, java]
+----
+AsyncSecurityCheckExecutor asyncCheckExecutor = securityService.getAsyncCheckExecutor();
+----
+
+===== Custom Self Defence Checks
```

Comment:
I suggest move this one and the above to its own section. Implementing custom self defence checks doesn't have anything to do with the executors.